add usage for eventually consistent replication (#6729)
pingcap/docs: ticdc/manage-ticdc.md

> + If your servers use mechanical hard drives or other storage devices that have high latency or limited bandwidth, use the unified sorter with caution.
> + The total free capacity of hard drives must be greater than or equal to 500G. If you need to replicate a large amount of historical data, make sure that the free capacity on each node is greater than or equal to the size of the incremental data that needs to be replicated.
> + Unified sorter is enabled by default. If your servers do not match the above requirements and you want to disable the unified sorter, you need to manually set `sort-engine` to `memory` for the changefeed.

> > + To enable Unified Sorter on an existing changefeed, see the methods provided in [How do I handle the OOM that occurs after TiCDC is restarted after a task interruption?](/ticdc/troubleshoot-ticdc.md#how-do-i-handle-the-oom-that-occurs-after-ticdc-is-restarted-after-a-task-interruption).

Reviewer: ```suggestion
> + To enable Unified Sorter on an existing changefeed, see the methods provided in [How do I handle the OOM that occurs after TiCDC is restarted after a task interruption?](/ticdc/troubleshoot-ticdc.md#what-should-i-do-to-handle-the-oom-that-occurs-after-ticdc-is-restarted-after-a-task-interruption).
```

This feature supports TiCDC to replicate incremental data from a TiDB cluster to the secondary relational database TiDB/Aurora/MySQL/MariaDB. In case the primary cluster crashes, TiCDC can recover the secondary cluster to a certain snapshot in the primary cluster within 30 minutes, given the condition that before disaster the replication status of TiCDC is normal and replication lag is small. It allows data loss of less than 5 minutes, that is, RPO <= 30min, and RTO <= 5min.

### Prerequisites 

- Prepare a highly available Amazon S3 storage or NFS system for storing TiCDC's real-time incremental data backup files. These files can be accessed in case of an primary cluster disaster.
- Enable this feature for changefeeds that need to have eventual consistency in disaster scenarios. To enable it, you can add the following configuration to the changefeed configuration file.

```toml
[consistent]

> # Consistency level. Options include.
> # - none: the default value. In a non-disaster scenario, eventual consistency is only guaranteed if and only if finished-ts is specified. 
> # - eventual: Use redo log to guarantee eventual consistency in case of the primary cluster disasters. 
> level = "eventual"
> # Individual redo log file size, in MiB. By default, it's 64. It is recommended to be no more than 128.
> max-log-size = 64
> # The interval for flushing or uploading redo logs to S3, in milliseconds. By default, it's 1000. The recommended range is 500-2000.
> flush-interval = 1000
> # Form of storing redo log, including nfs (NFS directory) and S3 (uploading to S3).

Reviewer: ```suggestion
# Consistency level. Options include:
# - none: the default value. In a non-disaster scenario, eventual consistency is only guaranteed if and only if finished-ts is specified. 
# - eventual: Uses redo log to guarantee eventual consistency in case of the primary cluster disasters. 
level = "eventual"

# Individual redo log file size, in MiB. By default, it's 64. It is recommended to be no more than 128.
max-log-size = 64

# The interval for flushing or uploading redo logs to S3, in milliseconds. By default, it's 1000. The recommended range is 500-2000.
flush-interval = 1000

# Form of storing redo log, including nfs (NFS directory) and S3 (uploading to S3).
```

storage = "s3://logbucket/test-changefeed?endpoint=http://$S3_ENDPOINT/"
```

### Disaster recovery

When a disaster happens in the primary cluster, you need to recover manually in the secondary cluster by running the `cdc redo` command. The recovery process is as follows.

1. Ensure that all the TiCDC processes have exited. This is to prevent the primary cluster from resuming service during data recovery and prevent TiCDC from restarting data synchronization.

> 2. Use cdc binary for data recovery. Run the following command.

Reviewer: ```suggestion
2. Use cdc binary for data recovery. Run the following command:
```

```shell
cdc redo apply --tmp-dir="/tmp/cdc/redo/apply" \
    --storage="s3://logbucket/test-changefeed?endpoint=http://10.0.10.25:24927/" \
    --sink-uri="mysql://normal:123456@10.0.10.55:3306/"
```

In this command:

> - `tmp-dir`: Specify the temporary directory for downloading TiCDC incremental data backup files.
> - `storage`: Specify the address for storing the TiCDC incremental data backup files, either an Amazon S3 storage or an NFS directory.
> - `sink-uri`: Specify the secondary cluster address to restore the data to. Scheme can only be `mysql`.

Reviewer: ```suggestion
- `tmp-dir`: Specifies the temporary directory for downloading TiCDC incremental data backup files.
- `storage`: Specifies the address for storing the TiCDC incremental data backup files, either an Amazon S3 storage or an NFS directory.
- `sink-uri`: Specifies the secondary cluster address to restore the data to. Scheme can only be `mysql`.
```

> + If your servers use mechanical hard drives or other storage devices that have high latency or limited bandwidth, use the unified sorter with caution.
> + The total free capacity of hard drives must be greater than or equal to 500G. If you need to replicate a large amount of historical data, make sure that the free capacity on each node is greater than or equal to the size of the incremental data that needs to be replicated.
> + Unified sorter is enabled by default. If your servers do not match the above requirements and you want to disable the unified sorter, you need to manually set `sort-engine` to `memory` for the changefeed.

> > + To enable Unified Sorter on an existing changefeed, see the methods provided in [How do I handle the OOM that occurs after TiCDC is restarted after a task interruption?](/ticdc/troubleshoot-ticdc.md#what-should-i-do-to-handle-the-oom-that-occurs-after-ticdc-is-restarted-after-a-task-interruption).

Reviewer 2: ```suggestion
> + To enable Unified Sorter on an existing changefeed, see the methods provided in [How do I handle the OOM that occurs after TiCDC is restarted after a task interruption?](/ticdc/troubleshoot-ticdc.md#what-should-i-do-to-handle-the-oom-that-occurs-after-ticdc-is-restarted-after-a-task-interruption). 
```

> + The total free capacity of hard drives must be greater than or equal to 500G. If you need to replicate a large amount of historical data, make sure that the free capacity on each node is greater than or equal to the size of the incremental data that needs to be replicated.
> + Unified sorter is enabled by default. If your servers do not match the above requirements and you want to disable the unified sorter, you need to manually set `sort-engine` to `memory` for the changefeed.
> + To enable Unified Sorter on an existing changefeed, see the methods provided in [How do I handle the OOM that occurs after TiCDC is restarted after a task interruption?](/ticdc/troubleshoot-ticdc.md#how-do-i-handle-the-oom-that-occurs-after-ticdc-is-restarted-after-a-task-interruption).

## Ultimate consistent replication in disaster scenarios

> Starting from v5.3.0, TiCDC provides the ultimate consistent replication capability in disaster scenarios. When a disaster occurs in the production cluster (i.e. TiCDC's primary cluster) and the service cannot be resumed in a short period of time, TiCDC needs to provide the ability to ensure the consistency of data in the secondary cluster. Meanwhile, TiCDC needs to allow the business to quickly switch the traffic to the secondary cluster to avoid the database being unavailable for a long time and affecting the business.

Author: ```suggestion
Starting from v5.3.0, TiCDC provides the eventually consistent replication capability in disaster scenarios. When a disaster occurs in the production cluster (i.e. TiCDC's primary cluster) and the service cannot be resumed in a short period of time, TiCDC needs to provide the ability to ensure the consistency of data in the secondary cluster. Meanwhile, TiCDC needs to allow the business to quickly switch the traffic to the secondary cluster to avoid the database being unavailable for a long time and affecting the business.
```

> This feature supports TiCDC to replicate incremental data from a TiDB cluster to the secondary relational database TiDB/Aurora/MySQL/MariaDB. In case the primary cluster crashes, TiCDC can recover the secondary cluster to a certain snapshot in the primary cluster within 30 minutes, given the condition that before disaster the replication status of TiCDC is normal and replication lag is small. . It allows data loss of less than 5 minutes, that is, RPO <= 30min, and RTO <= 5min.

Author: ```suggestion
This feature supports TiCDC to replicate incremental data from a TiDB cluster to the secondary relational database TiDB/Aurora/MySQL/MariaDB. In case the primary cluster crashes, TiCDC can recover the secondary cluster to a certain snapshot in the primary cluster within 30 minutes, given the condition that before disaster the replication status of TiCDC is normal and replication lag is small. It allows data loss of less than 5 minutes, that is, RPO <= 30min, and RTO <= 5min.
```

### Prerequisites 

- Prepare a highly available Amazon S3 storage or NFS system for storing TiCDC's real-time incremental data backup files. These files can be accessed in case of an upstream disaster.
- Enable this feature for changefeeds that need to have ultimate consistency in disaster scenarios. To enable it, you can add the following configuration to the changefeed configuration file.

```toml
[consistent]
# Consistency level. Options include.

> # - none: the default value. In a non-disaster scenario, ultimate consistency is only guaranteed if only finished-ts is specified. 

Reviewer 3: ```suggestion
# - none: the default value. In a non-disaster scenario, eventually consistency is only guaranteed if and only if finished-ts is specified. 
```

> # - eventual: Use redo log to guarantee ultimate consistency in case of upstream disasters. 

Reviewer 3: ```suggestion
# - eventual: Use redo log to guarantee eventual consistency in case of upstream disasters. 
```

> # - eventual: Use redo log to guarantee ultimate consistency in case of the primary cluster disasters. 

Author: ```suggestion
# - eventual: Use redo log to guarantee eventual consistency in case of the primary cluster disasters. 
```

level = "eventual"
# Individual redo log file size, in MiB. By default, it's 64. It is recommended to be no more than 128.
max-log-size = 64

> # The interval for refreshing or uploading redo logs to S3, in milliseconds. By default, it's 1000. The recommended range is 500-2000.

Reviewer 3: ```suggestion
# The interval for flushing or uploading redo logs to S3, in milliseconds. By default, it's 1000. The recommended range is 500-2000.
```

flush-interval = 1000
# Form of storing redo log, including nfs (NFS directory) and S3 (uploading to S3).
storage = "s3://logbucket/test-changefeed?endpoint=http://$S3_ENDPOINT/"
```

### Disaster recovery

When a disaster happens in upstream, you need to recover manually in downstream by running the `cdc redo` command. The recovery process is as follows.

> 1. Ensure that the TiCDC process has exited. This is to prevent the upstream from resuming service during data recovery and prevent TiCDC from restarting data synchronization.

Reviewer 3: ```suggestion
1. Ensure that all the TiCDC processes have exited. This is to prevent the upstream from resuming service during data recovery and prevent TiCDC from restarting data synchronization.
```
